influxql/functions.go

```diff
@@ -936,27 +936,11 @@ type firstLastMapOutput struct {
 
 // MapFirst collects the values to pass to the reducer
 func MapFirst(itr Iterator) interface{} {
-	out := &firstLastMapOutput{}
-	pointsYielded := false
-
-	for k, v := itr.Next(); k != -1; k, v = itr.Next() {
-		// Initialize first
-		if !pointsYielded {
-			out.Time = k
-			out.Val = v
-			pointsYielded = true
-		}
-		if k < out.Time {
-			out.Time = k
-			out.Val = v
-		} else if k == out.Time && greaterThan(v, out.Val) {
```

Comment:
Yes, I think this is necessary. You need to see *all* points at the first timestamp, so you can beak any tie in a predictable manner.